bentoml/cli/azure_functions.py

```diff
@@ -0,0 +1,432 @@
+)
+from bentoml.yatai.deployment.store import ALL_NAMESPACE_TAG
+from bentoml.exceptions import BentoMLException
+from bentoml.yatai.proto import status_pb2
+from bentoml.yatai.proto.deployment_pb2 import DeploymentSpec
+from bentoml.utils import status_pb_to_error_code_and_message
+from bentoml.utils.usage_stats import track_cli
+from bentoml.yatai.client import YataiClient
+
+PLATFORM_NAME = DeploymentSpec.DeploymentOperator.Name(DeploymentSpec.AZURE_FUNCTIONS)
+
+
+def get_azure_functions_sub_command():
+    # pylint: disable=unused-variable
+
+    @click.group(
+        name='azure-functions',
+        help='Commands for Azure Functions BentoService deployment',
+        cls=BentoMLCommandGroup,
+    )
+    def azure_functions():
+        pass
+
+    @azure_functions.command(help='Deploy BentoService to Azure Functions')
+    @click.option(
+        '-n',
+        '--namespace',
+        type=click.STRING,
+        help='Deployment namespace managed by BentoML, default value is "dev" which'
+        'can be changed in BentoML configuration yatai_service/default_namespace',
+    )
+    @click.argument('name', type=click.STRING)
+    @click.option(
+        '-b',
+        '--bento',
+        '--bento-service-bundle',
+        type=click.STRING,
+        required=True,
+        callback=parse_bento_tag_callback,
+        help='Target BentoService to be deployed, referenced by its name and version '
+        'in the format of name:version. For example: "iris_classifier:v1.2.0"',
+    )
+    @click.option(
+        '--location',
+        type=click.STRING,
+        help='The Azure location name for the deployment',
+        required=True,
+    )
+    @click.option(
+        '--min-instances',
+        type=click.INT,
+        default=DEFAULT_MIN_INSTANCE_COUNT,
+        help=f'The minimum number of workers for the deployment. The default value is '
+        f'{DEFAULT_MIN_INSTANCE_COUNT}',
+    )
+    @click.option(
+        '--max-burst',
+        type=click.INT,
+        default=DEFAULT_MAX_BURST,
+        help=f'The maximum number of elastic workers for the deployment. '
+        f'The default value is {DEFAULT_MAX_BURST}',
+    )
+    @click.option(
+        '--premium-plan-sku',
+        type=click.Choice(AZURE_FUNCTIONS_PREMIUM_PLAN_SKUS),
+        default=DEFAULT_PREMIUM_PLAN_SKU,
+        help=f'The Azure Functions premium SKU for the deployment. The default value '
+        f'is {DEFAULT_PREMIUM_PLAN_SKU}',
+    )
+    @click.option(
+        '-l',
+        '--labels',
+        type=click.STRING,
+        callback=parse_labels_callback,
+        help='Key:value pairs that are attached to deployments and intended to be used '
+        'to specify identifying attributes of the deployments that are meaningful to '
+        'users. Multiple labels are separated with `,`',
+    )
+    @click.option(
+        '--function-auth-level',
+        type=click.Choice(AZURE_FUNCTIONS_AUTH_LEVELS),
+        default=DEFAULT_FUNCTION_AUTH_LEVEL,
```

Comment:
remove default here since we have the default set on the server-side? it's better to just do it once in one place